// spinners-angular/__ivy_ngcc__/fesm2015/spinners-angular-spinner-infinity.js
import { Component, NgModule } from '@angular/core';
import { SpinnerComponentWithSecondaryColor } from 'spinners-angular/spinners-utils';
import { BrowserModule } from '@angular/platform-browser';

import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/common';

function SpinnerInfinityComponent__svg_svg_0_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵnamespaceSVG();
    ɵngcc0.ɵɵelementStart(0, "svg", 1);
    ɵngcc0.ɵɵelementStart(1, "defs");
    ɵngcc0.ɵɵelement(2, "path", 2);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelement(3, "use", 3);
    ɵngcc0.ɵɵelement(4, "use", 4);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵproperty("ngStyle", ctx_r0.svgStyle);
    ɵngcc0.ɵɵadvance(3);
    ɵngcc0.ɵɵattribute("stroke", ctx_r0.secondaryColor)("stroke-width", ctx_r0.strokeWidth);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngStyle", ctx_r0.dashStyle);
    ɵngcc0.ɵɵattribute("stroke-width", ctx_r0.strokeWidth);
} }
class SpinnerInfinityComponent extends SpinnerComponentWithSecondaryColor {
    get strokeWidth() {
        return 7 * (this.thickness / 100);
    }
    get dashStyle() {
        return !this.still
            ? { animation: `spinners-angular-infinity ${140 / this.speed}s linear infinite` }
            : {};
    }
}
SpinnerInfinityComponent.ɵfac = /*@__PURE__*/ function () { let ɵSpinnerInfinityComponent_BaseFactory; return function SpinnerInfinityComponent_Factory(t) { return (ɵSpinnerInfinityComponent_BaseFactory || (ɵSpinnerInfinityComponent_BaseFactory = ɵngcc0.ɵɵgetInheritedFactory(SpinnerInfinityComponent)))(t || SpinnerInfinityComponent); }; }();
SpinnerInfinityComponent.ɵcmp = /*@__PURE__*/ ɵngcc0.ɵɵdefineComponent({ type: SpinnerInfinityComponent, selectors: [["sa-spinner-infinity"]], features: [ɵngcc0.ɵɵInheritDefinitionFeature], decls: 1, vars: 1, consts: [["fill", "none", "viewBox", "0 0 131 55", 3, "ngStyle", 4, "ngIf"], ["fill", "none", "viewBox", "0 0 131 55", 3, "ngStyle"], ["d", "M46.57 45.5138C36.346 55.4954 19.8919 55.4954 9.66794 45.5138C-0.55598 35.5321 -0.55598 19.4678 9.66794 9.48624C19.8919 -0.495412 36.346 -0.495412 46.57 9.48624L84.4303 45.5138C94.6543 55.4954 111.108 55.4954 121.332 45.5138C131.556 35.5321 131.556 19.4678 121.332 9.48624C111.108 -0.495412 94.6543 -0.495412 84.4303 9.48624L46.57 45.5138Z", "id", "spinners-react-infinity-path"], [0, "xlink", "href", "#spinners-react-infinity-path"], ["fill", "none", "stroke", "currentColor", "stroke-dasharray", "1, 347", "stroke-dashoffset", "75", "stroke-linecap", "round", 0, "xlink", "href", "#spinners-react-infinity-path", 3, "ngStyle"]], template: function SpinnerInfinityComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵtemplate(0, SpinnerInfinityComponent__svg_svg_0_Template, 5, 5, "svg", 0);
    } if (rf & 2) {
        ɵngcc0.ɵɵproperty("ngIf", ctx.enabled);
    } }, directives: [ɵngcc1.NgIf, ɵngcc1.NgStyle], styles: ["@-webkit-keyframes spinners-angular-infinity{0%{stroke-dasharray:1,347;stroke-dashoffset:75}25%,75%{stroke-dasharray:17,330}50%{stroke-dasharray:1,347}to{stroke-dasharray:1,347;stroke-dashoffset:423}}@keyframes spinners-angular-infinity{0%{stroke-dasharray:1,347;stroke-dashoffset:75}25%,75%{stroke-dasharray:17,330}50%{stroke-dasharray:1,347}to{stroke-dasharray:1,347;stroke-dashoffset:423}}"] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(SpinnerInfinityComponent, [{
        type: Component,
        args: [{
                selector: 'sa-spinner-infinity',
                template: "<svg *ngIf=\"enabled\" fill=\"none\" [ngStyle]=\"svgStyle\" viewBox=\"0 0 131 55\">\r\n  <defs>\r\n    <path\r\n      d=\"M46.57 45.5138C36.346 55.4954 19.8919 55.4954 9.66794 45.5138C-0.55598 35.5321 -0.55598 19.4678 9.66794 9.48624C19.8919 -0.495412 36.346 -0.495412 46.57 9.48624L84.4303 45.5138C94.6543 55.4954 111.108 55.4954 121.332 45.5138C131.556 35.5321 131.556 19.4678 121.332 9.48624C111.108 -0.495412 94.6543 -0.495412 84.4303 9.48624L46.57 45.5138Z\"\r\n      id=\"spinners-react-infinity-path\"\r\n    />\r\n  </defs>\r\n  <use\r\n    [attr.stroke]=\"secondaryColor\"\r\n    [attr.stroke-width]=\"strokeWidth\"\r\n    xlink:href=\"#spinners-react-infinity-path\"\r\n  />\r\n  <use\r\n    fill=\"none\"\r\n    stroke=\"currentColor\"\r\n    stroke-dasharray=\"1, 347\"\r\n    stroke-dashoffset=\"75\"\r\n    stroke-linecap=\"round\"\r\n    [attr.stroke-width]=\"strokeWidth\"\r\n    [ngStyle]=\"dashStyle\"\r\n    xlink:href=\"#spinners-react-infinity-path\"\r\n  />\r\n</svg>\r\n",
                styles: ["@-webkit-keyframes spinners-angular-infinity{0%{stroke-dasharray:1,347;stroke-dashoffset:75}25%,75%{stroke-dasharray:17,330}50%{stroke-dasharray:1,347}to{stroke-dasharray:1,347;stroke-dashoffset:423}}@keyframes spinners-angular-infinity{0%{stroke-dasharray:1,347;stroke-dashoffset:75}25%,75%{stroke-dasharray:17,330}50%{stroke-dasharray:1,347}to{stroke-dasharray:1,347;stroke-dashoffset:423}}\n"]
            }]
    }], null, null); })();

class SpinnerInfinityModule {
}
SpinnerInfinityModule.ɵfac = function SpinnerInfinityModule_Factory(t) { return new (t || SpinnerInfinityModule)(); };
SpinnerInfinityModule.ɵmod = /*@__PURE__*/ ɵngcc0.ɵɵdefineNgModule({ type: SpinnerInfinityModule });
SpinnerInfinityModule.ɵinj = /*@__PURE__*/ ɵngcc0.ɵɵdefineInjector({ imports: [[BrowserModule]] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(SpinnerInfinityModule, [{
        type: NgModule,
        args: [{
                imports: [BrowserModule],
                declarations: [SpinnerInfinityComponent],
                exports: [SpinnerInfinityComponent]
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(SpinnerInfinityModule, { declarations: function () { return [SpinnerInfinityComponent]; }, imports: function () { return [BrowserModule]; }, exports: function () { return [SpinnerInfinityComponent]; } }); })();

/**
 * Generated bundle index. Do not edit.
 */

export { SpinnerInfinityComponent, SpinnerInfinityModule };
