// spinners-angular/__ivy_ngcc__/fesm2015/spinners-angular-spinner-diamond.js
import { Component, NgModule } from '@angular/core';
import { SpinnerComponentWithSecondaryColor } from 'spinners-angular/spinners-utils';
import { BrowserModule } from '@angular/platform-browser';

import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/common';

function SpinnerDiamondComponent__svg_svg_0__svg_circle_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵnamespaceSVG();
    ɵngcc0.ɵɵelement(0, "circle");
} if (rf & 2) {
    const c_r3 = ctx.$implicit;
    const ctx_r1 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵattribute("cx", c_r3.x)("cy", c_r3.y)("r", 2.5 * (ctx_r1.thickness / 100))("fill", ctx_r1.secondaryColor);
} }
function SpinnerDiamondComponent__svg_svg_0__svg_circle_3_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵnamespaceSVG();
    ɵngcc0.ɵɵelement(0, "circle");
} if (rf & 2) {
    const c_r4 = ctx.$implicit;
    const ctx_r2 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵattribute("r", 3.5 * (ctx_r2.thickness / 100))("cx", c_r4.x)("cy", c_r4.y);
} }
function SpinnerDiamondComponent__svg_svg_0_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵnamespaceSVG();
    ɵngcc0.ɵɵelementStart(0, "svg", 1);
    ɵngcc0.ɵɵtemplate(1, SpinnerDiamondComponent__svg_svg_0__svg_circle_1_Template, 1, 4, "circle", 2);
    ɵngcc0.ɵɵelementStart(2, "g", 3);
    ɵngcc0.ɵɵtemplate(3, SpinnerDiamondComponent__svg_svg_0__svg_circle_3_Template, 1, 3, "circle", 2);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵproperty("ngStyle", ctx_r0.svgStyle);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngForOf", ctx_r0.coords);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngStyle", ctx_r0.groupStyle);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngForOf", ctx_r0.colorCoords);
} }
class SpinnerDiamondComponent extends SpinnerComponentWithSecondaryColor {
    constructor() {
        super(...arguments);
        this.colorCoords = [
            { x: 3, y: 48 },
            { x: 18, y: 33 },
            { x: 18, y: 48 },
            { x: 18, y: 63 },
            { x: 33, y: 48 },
        ];
    }
    get coords() {
        return [
            ...this.colorCoords,
            { x: 33, y: 18 },
            { x: 33, y: 33 },
            { x: 33, y: 63 },
            { x: 33, y: 78 },
            { x: 49, y: 3 },
            { x: 49, y: 18 },
            { x: 49, y: 33 },
            { x: 49, y: 48 },
            { x: 49, y: 63 },
            { x: 49, y: 78 },
            { x: 49, y: 93 },
            { x: 64, y: 18 },
            { x: 64, y: 33 },
            { x: 64, y: 48 },
            { x: 64, y: 63 },
            { x: 64, y: 78 },
            { x: 79, y: 33 },
            { x: 79, y: 48 },
            { x: 79, y: 63 },
            { x: 94, y: 48 },
        ];
    }
    get groupStyle() {
        return {
            animation: !this.still ? `spinners-angular-diamond ${140 / this.speed}s steps(2, end) infinite` : 'none',
        };
    }
}
SpinnerDiamondComponent.ɵfac = /*@__PURE__*/ function () { let ɵSpinnerDiamondComponent_BaseFactory; return function SpinnerDiamondComponent_Factory(t) { return (ɵSpinnerDiamondComponent_BaseFactory || (ɵSpinnerDiamondComponent_BaseFactory = ɵngcc0.ɵɵgetInheritedFactory(SpinnerDiamondComponent)))(t || SpinnerDiamondComponent); }; }();
SpinnerDiamondComponent.ɵcmp = /*@__PURE__*/ ɵngcc0.ɵɵdefineComponent({ type: SpinnerDiamondComponent, selectors: [["sa-spinner-diamond"]], features: [ɵngcc0.ɵɵInheritDefinitionFeature], decls: 1, vars: 1, consts: [["fill", "currentColor", "viewBox", "0 0 96 96", 3, "ngStyle", 4, "ngIf"], ["fill", "currentColor", "viewBox", "0 0 96 96", 3, "ngStyle"], [4, "ngFor", "ngForOf"], [3, "ngStyle"]], template: function SpinnerDiamondComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵtemplate(0, SpinnerDiamondComponent__svg_svg_0_Template, 4, 4, "svg", 0);
    } if (rf & 2) {
        ɵngcc0.ɵɵproperty("ngIf", ctx.enabled);
    } }, directives: [ɵngcc1.NgIf, ɵngcc1.NgStyle, ɵngcc1.NgForOf], styles: ["@-webkit-keyframes spinners-angular-diamond{25%{transform:translate(30px,-30px)}50%{transform:translate(60px)}75%{transform:translate(30px,30px)}}@keyframes spinners-angular-diamond{25%{transform:translate(30px,-30px)}50%{transform:translate(60px)}75%{transform:translate(30px,30px)}}"] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(SpinnerDiamondComponent, [{
        type: Component,
        args: [{
                selector: 'sa-spinner-diamond',
                template: "<svg *ngIf=\"enabled\" fill=\"currentColor\" [ngStyle]=\"svgStyle\" viewBox=\"0 0 96 96\">\r\n  <circle\r\n    *ngFor=\"let c of coords\"\r\n    [attr.cx]=\"c.x\"\r\n    [attr.cy]=\"c.y\"\r\n    [attr.r]=\"2.5 * (thickness / 100)\"\r\n    [attr.fill]=\"secondaryColor\"\r\n  />\r\n  <g [ngStyle]=\"groupStyle\">\r\n    <circle\r\n      *ngFor=\"let c of colorCoords; index as i\"\r\n      [attr.r]=\"3.5 * (thickness / 100)\"\r\n      [attr.cx]=\"c.x\"\r\n      [attr.cy]=\"c.y\"\r\n    />\r\n  </g>\r\n</svg>\r\n",
                styles: ["@-webkit-keyframes spinners-angular-diamond{25%{transform:translate(30px,-30px)}50%{transform:translate(60px)}75%{transform:translate(30px,30px)}}@keyframes spinners-angular-diamond{25%{transform:translate(30px,-30px)}50%{transform:translate(60px)}75%{transform:translate(30px,30px)}}\n"]
            }]
    }], null, null); })();

class SpinnerDiamondModule {
}
SpinnerDiamondModule.ɵfac = function SpinnerDiamondModule_Factory(t) { return new (t || SpinnerDiamondModule)(); };
SpinnerDiamondModule.ɵmod = /*@__PURE__*/ ɵngcc0.ɵɵdefineNgModule({ type: SpinnerDiamondModule });
SpinnerDiamondModule.ɵinj = /*@__PURE__*/ ɵngcc0.ɵɵdefineInjector({ imports: [[BrowserModule]] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(SpinnerDiamondModule, [{
        type: NgModule,
        args: [{
                imports: [BrowserModule],
                declarations: [SpinnerDiamondComponent],
                exports: [SpinnerDiamondComponent]
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(SpinnerDiamondModule, { declarations: function () { return [SpinnerDiamondComponent]; }, imports: function () { return [BrowserModule]; }, exports: function () { return [SpinnerDiamondComponent]; } }); })();

/**
 * Generated bundle index. Do not edit.
 */

export { SpinnerDiamondComponent, SpinnerDiamondModule };
