// spinners-angular/__ivy_ngcc__/fesm2015/spinners-angular-spinner-round.js
import { Component, NgModule } from '@angular/core';
import { SpinnerComponent } from 'spinners-angular/spinners-utils';
import { BrowserModule } from '@angular/platform-browser';

import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/common';

function SpinnerRoundComponent__svg_svg_0_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵnamespaceSVG();
    ɵngcc0.ɵɵelementStart(0, "svg", 1);
    ɵngcc0.ɵɵelement(1, "circle", 2);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵproperty("ngStyle", ctx_r0.svgStyle);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngStyle", ctx_r0.circleStyle);
    ɵngcc0.ɵɵattribute("stroke-width", ctx_r0.strokeWidth);
} }
class SpinnerRoundComponent extends SpinnerComponent {
    get strokeWidth() {
        return 3 * (this.thickness / 100);
    }
    get circleStyle() {
        return {
            animation: !this.still ? `spinners-angular-round ${140 / this.speed}s ease-in-out infinite` : 'none',
            transformOrigin: 'center',
        };
    }
}
SpinnerRoundComponent.ɵfac = /*@__PURE__*/ function () { let ɵSpinnerRoundComponent_BaseFactory; return function SpinnerRoundComponent_Factory(t) { return (ɵSpinnerRoundComponent_BaseFactory || (ɵSpinnerRoundComponent_BaseFactory = ɵngcc0.ɵɵgetInheritedFactory(SpinnerRoundComponent)))(t || SpinnerRoundComponent); }; }();
SpinnerRoundComponent.ɵcmp = /*@__PURE__*/ ɵngcc0.ɵɵdefineComponent({ type: SpinnerRoundComponent, selectors: [["sa-spinner-round"]], features: [ɵngcc0.ɵɵInheritDefinitionFeature], decls: 1, vars: 1, consts: [["fill", "none", "viewBox", "0 0 66 66", 3, "ngStyle", 4, "ngIf"], ["fill", "none", "viewBox", "0 0 66 66", 3, "ngStyle"], ["cx", "33", "cy", "33", "fill", "none", "r", "28", "stroke", "currentColor", 3, "ngStyle"]], template: function SpinnerRoundComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵtemplate(0, SpinnerRoundComponent__svg_svg_0_Template, 2, 3, "svg", 0);
    } if (rf & 2) {
        ɵngcc0.ɵɵproperty("ngIf", ctx.enabled);
    } }, directives: [ɵngcc1.NgIf, ɵngcc1.NgStyle], styles: ["@-webkit-keyframes spinners-angular-round{27%,73%{r:2px;stroke-width:4}}@keyframes spinners-angular-round{27%,73%{r:2px;stroke-width:4}}"] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(SpinnerRoundComponent, [{
        type: Component,
        args: [{
                selector: 'sa-spinner-round',
                template: "<svg *ngIf=\"enabled\" fill=\"none\" [ngStyle]=\"svgStyle\" viewBox=\"0 0 66 66\">\r\n  <circle\r\n    cx=\"33\"\r\n    cy=\"33\"\r\n    fill=\"none\"\r\n    r=\"28\"\r\n    stroke=\"currentColor\"\r\n    [attr.stroke-width]=\"strokeWidth\"\r\n    [ngStyle]=\"circleStyle\"\r\n  />\r\n</svg>\r\n",
                styles: ["@-webkit-keyframes spinners-angular-round{27%,73%{r:2px;stroke-width:4}}@keyframes spinners-angular-round{27%,73%{r:2px;stroke-width:4}}\n"]
            }]
    }], null, null); })();

class SpinnerRoundModule {
}
SpinnerRoundModule.ɵfac = function SpinnerRoundModule_Factory(t) { return new (t || SpinnerRoundModule)(); };
SpinnerRoundModule.ɵmod = /*@__PURE__*/ ɵngcc0.ɵɵdefineNgModule({ type: SpinnerRoundModule });
SpinnerRoundModule.ɵinj = /*@__PURE__*/ ɵngcc0.ɵɵdefineInjector({ imports: [[BrowserModule]] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(SpinnerRoundModule, [{
        type: NgModule,
        args: [{
                imports: [BrowserModule],
                declarations: [SpinnerRoundComponent],
                exports: [SpinnerRoundComponent]
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(SpinnerRoundModule, { declarations: function () { return [SpinnerRoundComponent]; }, imports: function () { return [BrowserModule]; }, exports: function () { return [SpinnerRoundComponent]; } }); })();

/**
 * Generated bundle index. Do not edit.
 */

export { SpinnerRoundComponent, SpinnerRoundModule };
