// spinners-angular/__ivy_ngcc__/fesm2015/spinners-angular-spinner-circular-split.js
import { Component, NgModule } from '@angular/core';
import { SpinnerComponentWithSecondaryColor } from 'spinners-angular/spinners-utils';
import { BrowserModule } from '@angular/platform-browser';

import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/common';

function SpinnerCircularSplitComponent__svg_svg_0_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵnamespaceSVG();
    ɵngcc0.ɵɵelementStart(0, "svg", 1);
    ɵngcc0.ɵɵelement(1, "circle", 2);
    ɵngcc0.ɵɵelement(2, "circle", 3);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵproperty("ngStyle", ctx_r0.svgStyle);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵattribute("stroke", ctx_r0.secondaryColor)("stroke-width", ctx_r0.strokeWidth);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngStyle", ctx_r0.dashStyle);
    ɵngcc0.ɵɵattribute("stroke-width", ctx_r0.strokeWidth);
} }
class SpinnerCircularSplitComponent extends SpinnerComponentWithSecondaryColor {
    get strokeWidth() {
        return 4 * (this.thickness / 100);
    }
    get dashStyle() {
        return Object.assign({ color: this.color }, (!this.still
            ? { animation: `spinners-angular-circular-split ${140 / this.speed}s linear infinite` }
            : {}));
    }
}
SpinnerCircularSplitComponent.ɵfac = /*@__PURE__*/ function () { let ɵSpinnerCircularSplitComponent_BaseFactory; return function SpinnerCircularSplitComponent_Factory(t) { return (ɵSpinnerCircularSplitComponent_BaseFactory || (ɵSpinnerCircularSplitComponent_BaseFactory = ɵngcc0.ɵɵgetInheritedFactory(SpinnerCircularSplitComponent)))(t || SpinnerCircularSplitComponent); }; }();
SpinnerCircularSplitComponent.ɵcmp = /*@__PURE__*/ ɵngcc0.ɵɵdefineComponent({ type: SpinnerCircularSplitComponent, selectors: [["sa-spinner-circular-split"]], features: [ɵngcc0.ɵɵInheritDefinitionFeature], decls: 1, vars: 1, consts: [["fill", "none", "viewBox", "0 0 66 66", 3, "ngStyle", 4, "ngIf"], ["fill", "none", "viewBox", "0 0 66 66", 3, "ngStyle"], ["cx", "33", "cy", "33", "fill", "none", "r", "28"], ["cx", "33", "cy", "33", "fill", "none", "r", "28", "stroke", "currentColor", "stroke-dasharray", "5, 170", "stroke-dashoffset", "1", "stroke-linecap", "round", "transform", "rotate(-90 33 33)", 3, "ngStyle"]], template: function SpinnerCircularSplitComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵtemplate(0, SpinnerCircularSplitComponent__svg_svg_0_Template, 3, 5, "svg", 0);
    } if (rf & 2) {
        ɵngcc0.ɵɵproperty("ngIf", ctx.enabled);
    } }, directives: [ɵngcc1.NgIf, ɵngcc1.NgStyle], styles: ["@-webkit-keyframes spinners-angular-circular-split{0%{stroke-dashoffset:1;stroke-dasharray:5,170}10%{stroke-dashoffset:13;stroke-dasharray:30,145}13%{stroke-dashoffset:-11;stroke-dasharray:5,145}50%{stroke-dasharray:5,0,5,165;stroke-dashoffset:-82}51%{stroke-dasharray:2,0,2,139;stroke-dashoffset:-85}61%{stroke-dasharray:15,0,15,165;stroke-dashoffset:-72}64%{stroke-dasharray:5,20,5,145;stroke-dashoffset:-72}}@keyframes spinners-angular-circular-split{0%{stroke-dashoffset:1;stroke-dasharray:5,170}10%{stroke-dashoffset:13;stroke-dasharray:30,145}13%{stroke-dashoffset:-11;stroke-dasharray:5,145}50%{stroke-dasharray:5,0,5,165;stroke-dashoffset:-82}51%{stroke-dasharray:2,0,2,139;stroke-dashoffset:-85}61%{stroke-dasharray:15,0,15,165;stroke-dashoffset:-72}64%{stroke-dasharray:5,20,5,145;stroke-dashoffset:-72}}"] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(SpinnerCircularSplitComponent, [{
        type: Component,
        args: [{
                selector: 'sa-spinner-circular-split',
                template: "<svg *ngIf=\"enabled\" fill=\"none\" [ngStyle]=\"svgStyle\" viewBox=\"0 0 66 66\">\r\n  <circle\r\n    cx=\"33\"\r\n    cy=\"33\"\r\n    fill=\"none\"\r\n    r=\"28\"\r\n    [attr.stroke]=\"secondaryColor\"\r\n    [attr.stroke-width]=\"strokeWidth\"\r\n  />\r\n  <circle\r\n    cx=\"33\"\r\n    cy=\"33\"\r\n    fill=\"none\"\r\n    r=\"28\"\r\n    stroke=\"currentColor\"\r\n    stroke-dasharray=\"5, 170\"\r\n    stroke-dashoffset=\"1\"\r\n    stroke-linecap=\"round\"\r\n    transform=\"rotate(-90 33 33)\"\r\n    [attr.stroke-width]=\"strokeWidth\"\r\n    [ngStyle]=\"dashStyle\"\r\n  />\r\n</svg>\r\n",
                styles: ["@-webkit-keyframes spinners-angular-circular-split{0%{stroke-dashoffset:1;stroke-dasharray:5,170}10%{stroke-dashoffset:13;stroke-dasharray:30,145}13%{stroke-dashoffset:-11;stroke-dasharray:5,145}50%{stroke-dasharray:5,0,5,165;stroke-dashoffset:-82}51%{stroke-dasharray:2,0,2,139;stroke-dashoffset:-85}61%{stroke-dasharray:15,0,15,165;stroke-dashoffset:-72}64%{stroke-dasharray:5,20,5,145;stroke-dashoffset:-72}}@keyframes spinners-angular-circular-split{0%{stroke-dashoffset:1;stroke-dasharray:5,170}10%{stroke-dashoffset:13;stroke-dasharray:30,145}13%{stroke-dashoffset:-11;stroke-dasharray:5,145}50%{stroke-dasharray:5,0,5,165;stroke-dashoffset:-82}51%{stroke-dasharray:2,0,2,139;stroke-dashoffset:-85}61%{stroke-dasharray:15,0,15,165;stroke-dashoffset:-72}64%{stroke-dasharray:5,20,5,145;stroke-dashoffset:-72}}\n"]
            }]
    }], null, null); })();

class SpinnerCircularSplitModule {
}
SpinnerCircularSplitModule.ɵfac = function SpinnerCircularSplitModule_Factory(t) { return new (t || SpinnerCircularSplitModule)(); };
SpinnerCircularSplitModule.ɵmod = /*@__PURE__*/ ɵngcc0.ɵɵdefineNgModule({ type: SpinnerCircularSplitModule });
SpinnerCircularSplitModule.ɵinj = /*@__PURE__*/ ɵngcc0.ɵɵdefineInjector({ imports: [[BrowserModule]] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(SpinnerCircularSplitModule, [{
        type: NgModule,
        args: [{
                imports: [BrowserModule],
                declarations: [SpinnerCircularSplitComponent],
                exports: [SpinnerCircularSplitComponent]
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(SpinnerCircularSplitModule, { declarations: function () { return [SpinnerCircularSplitComponent]; }, imports: function () { return [BrowserModule]; }, exports: function () { return [SpinnerCircularSplitComponent]; } }); })();

/**
 * Generated bundle index. Do not edit.
 */

export { SpinnerCircularSplitComponent, SpinnerCircularSplitModule };
