// spinners-angular/__ivy_ngcc__/fesm2015/spinners-angular-spinner-circular.js
import { Component, NgModule } from '@angular/core';
import { SpinnerComponentWithSecondaryColor } from 'spinners-angular/spinners-utils';
import { BrowserModule } from '@angular/platform-browser';

import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/common';

function SpinnerCircularComponent__svg_svg_0_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵnamespaceSVG();
    ɵngcc0.ɵɵelementStart(0, "svg", 1);
    ɵngcc0.ɵɵelement(1, "circle", 2);
    ɵngcc0.ɵɵelement(2, "circle", 3);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵproperty("ngStyle", ctx_r0.svgStyle);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵattribute("stroke", ctx_r0.secondaryColor)("stroke-width", ctx_r0.strokeWidth);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngStyle", ctx_r0.dashStyle);
    ɵngcc0.ɵɵattribute("stroke-width", ctx_r0.strokeWidth);
} }
class SpinnerCircularComponent extends SpinnerComponentWithSecondaryColor {
    get strokeWidth() {
        return 4 * (this.thickness / 100);
    }
    get dashStyle() {
        return Object.assign({ color: this.color }, (!this.still
            ? { animation: `spinners-angular-circular ${140 / this.speed}s linear infinite` }
            : {}));
    }
}
SpinnerCircularComponent.ɵfac = /*@__PURE__*/ function () { let ɵSpinnerCircularComponent_BaseFactory; return function SpinnerCircularComponent_Factory(t) { return (ɵSpinnerCircularComponent_BaseFactory || (ɵSpinnerCircularComponent_BaseFactory = ɵngcc0.ɵɵgetInheritedFactory(SpinnerCircularComponent)))(t || SpinnerCircularComponent); }; }();
SpinnerCircularComponent.ɵcmp = /*@__PURE__*/ ɵngcc0.ɵɵdefineComponent({ type: SpinnerCircularComponent, selectors: [["sa-spinner-circular"]], features: [ɵngcc0.ɵɵInheritDefinitionFeature], decls: 1, vars: 1, consts: [["fill", "none", "viewBox", "0 0 66 66", 3, "ngStyle", 4, "ngIf"], ["fill", "none", "viewBox", "0 0 66 66", 3, "ngStyle"], ["cx", "33", "cy", "33", "fill", "none", "r", "28"], ["cx", "33", "cy", "33", "fill", "none", "r", "28", "stroke", "currentColor", "stroke-dasharray", "1, 174", "stroke-dashoffset", "306", "stroke-linecap", "round", 3, "ngStyle"]], template: function SpinnerCircularComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵtemplate(0, SpinnerCircularComponent__svg_svg_0_Template, 3, 5, "svg", 0);
    } if (rf & 2) {
        ɵngcc0.ɵɵproperty("ngIf", ctx.enabled);
    } }, directives: [ɵngcc1.NgIf, ɵngcc1.NgStyle], styles: ["@-webkit-keyframes spinners-angular-circular{0%{stroke-dashoffset:306}50%{stroke-dasharray:40,134}to{stroke-dasharray:1,174;stroke-dashoffset:132}}@keyframes spinners-angular-circular{0%{stroke-dashoffset:306}50%{stroke-dasharray:40,134}to{stroke-dasharray:1,174;stroke-dashoffset:132}}"] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(SpinnerCircularComponent, [{
        type: Component,
        args: [{
                selector: 'sa-spinner-circular',
                template: "<svg *ngIf=\"enabled\" fill=\"none\" [ngStyle]=\"svgStyle\" viewBox=\"0 0 66 66\">\r\n  <circle\r\n    cx=\"33\"\r\n    cy=\"33\"\r\n    fill=\"none\"\r\n    r=\"28\"\r\n    [attr.stroke]=\"secondaryColor\"\r\n    [attr.stroke-width]=\"strokeWidth\"\r\n  />\r\n  <circle\r\n    cx=\"33\"\r\n    cy=\"33\"\r\n    fill=\"none\"\r\n    r=\"28\"\r\n    stroke=\"currentColor\"\r\n    stroke-dasharray=\"1, 174\"\r\n    stroke-dashoffset=\"306\"\r\n    stroke-linecap=\"round\"\r\n    [attr.stroke-width]=\"strokeWidth\"\r\n    [ngStyle]=\"dashStyle\"\r\n  />\r\n</svg>\r\n",
                styles: ["@-webkit-keyframes spinners-angular-circular{0%{stroke-dashoffset:306}50%{stroke-dasharray:40,134}to{stroke-dasharray:1,174;stroke-dashoffset:132}}@keyframes spinners-angular-circular{0%{stroke-dashoffset:306}50%{stroke-dasharray:40,134}to{stroke-dasharray:1,174;stroke-dashoffset:132}}\n"]
            }]
    }], null, null); })();

class SpinnerCircularModule {
}
SpinnerCircularModule.ɵfac = function SpinnerCircularModule_Factory(t) { return new (t || SpinnerCircularModule)(); };
SpinnerCircularModule.ɵmod = /*@__PURE__*/ ɵngcc0.ɵɵdefineNgModule({ type: SpinnerCircularModule });
SpinnerCircularModule.ɵinj = /*@__PURE__*/ ɵngcc0.ɵɵdefineInjector({ imports: [[BrowserModule]] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(SpinnerCircularModule, [{
        type: NgModule,
        args: [{
                imports: [BrowserModule],
                declarations: [SpinnerCircularComponent],
                exports: [SpinnerCircularComponent]
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(SpinnerCircularModule, { declarations: function () { return [SpinnerCircularComponent]; }, imports: function () { return [BrowserModule]; }, exports: function () { return [SpinnerCircularComponent]; } }); })();

/**
 * Generated bundle index. Do not edit.
 */

export { SpinnerCircularComponent, SpinnerCircularModule };
