// spinners-angular/__ivy_ngcc__/fesm2015/spinners-angular-spinner-round-outlined.js
import { Component, NgModule } from '@angular/core';
import { SpinnerComponent } from 'spinners-angular/spinners-utils';
import { BrowserModule } from '@angular/platform-browser';

import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/common';

function SpinnerRoundOutlinedComponent__svg_svg_0__svg_circle_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵnamespaceSVG();
    ɵngcc0.ɵɵelement(0, "circle", 3);
} if (rf & 2) {
    const animation_r2 = ctx.$implicit;
    const i_r3 = ctx.index;
    const ctx_r1 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵproperty("ngStyle", ctx_r1.circleStyle(animation_r2));
    ɵngcc0.ɵɵattribute("stroke-width", ctx_r1.strokeWidth(i_r3))("r", animation_r2.r);
} }
function SpinnerRoundOutlinedComponent__svg_svg_0_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵnamespaceSVG();
    ɵngcc0.ɵɵelementStart(0, "svg", 1);
    ɵngcc0.ɵɵtemplate(1, SpinnerRoundOutlinedComponent__svg_svg_0__svg_circle_1_Template, 1, 3, "circle", 2);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵproperty("ngStyle", ctx_r0.svgStyle);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngForOf", ctx_r0.animations);
} }
class SpinnerRoundOutlinedComponent extends SpinnerComponent {
    constructor() {
        super(...arguments);
        this.animations = [
            { r: 2 },
            {
                name: 'spinners-angular-round-outlined',
                r: 14,
            },
            {
                name: 'spinners-angular-round-outlined',
                r: 28,
            },
        ];
    }
    strokeWidth(index) {
        return index ? 3 * (this.thickness / 100) : 4;
    }
    circleStyle(animation) {
        return animation.name && !this.still
            ? { animation: `${animation.name} ${140 / this.speed}s ease-in-out infinite` }
            : {};
    }
}
SpinnerRoundOutlinedComponent.ɵfac = /*@__PURE__*/ function () { let ɵSpinnerRoundOutlinedComponent_BaseFactory; return function SpinnerRoundOutlinedComponent_Factory(t) { return (ɵSpinnerRoundOutlinedComponent_BaseFactory || (ɵSpinnerRoundOutlinedComponent_BaseFactory = ɵngcc0.ɵɵgetInheritedFactory(SpinnerRoundOutlinedComponent)))(t || SpinnerRoundOutlinedComponent); }; }();
SpinnerRoundOutlinedComponent.ɵcmp = /*@__PURE__*/ ɵngcc0.ɵɵdefineComponent({ type: SpinnerRoundOutlinedComponent, selectors: [["sa-spinner-round-outlined"]], features: [ɵngcc0.ɵɵInheritDefinitionFeature], decls: 1, vars: 1, consts: [["fill", "none", "viewBox", "0 0 66 66", 3, "ngStyle", 4, "ngIf"], ["fill", "none", "viewBox", "0 0 66 66", 3, "ngStyle"], ["cx", "33", "cy", "33", "fill", "none", "stroke", "currentColor", 3, "ngStyle", 4, "ngFor", "ngForOf"], ["cx", "33", "cy", "33", "fill", "none", "stroke", "currentColor", 3, "ngStyle"]], template: function SpinnerRoundOutlinedComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵtemplate(0, SpinnerRoundOutlinedComponent__svg_svg_0_Template, 2, 2, "svg", 0);
    } if (rf & 2) {
        ɵngcc0.ɵɵproperty("ngIf", ctx.enabled);
    } }, directives: [ɵngcc1.NgIf, ɵngcc1.NgStyle, ɵngcc1.NgForOf], styles: ["@-webkit-keyframes spinners-angular-round-outlined{35%,65%{r:2px;stroke-width:4}}@keyframes spinners-angular-round-outlined{35%,65%{r:2px;stroke-width:4}}"] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(SpinnerRoundOutlinedComponent, [{
        type: Component,
        args: [{
                selector: 'sa-spinner-round-outlined',
                template: "<svg *ngIf=\"enabled\" fill=\"none\" [ngStyle]=\"svgStyle\" viewBox=\"0 0 66 66\">\r\n  <circle\r\n    *ngFor=\"let animation of animations; index as i\"\r\n    cx=\"33\"\r\n    cy=\"33\"\r\n    fill=\"none\"\r\n    stroke=\"currentColor\"\r\n    [attr.stroke-width]=\"strokeWidth(i)\"\r\n    [attr.r]=\"animation.r\"\r\n    [ngStyle]=\"circleStyle(animation)\"\r\n  />\r\n</svg>\r\n",
                styles: ["@-webkit-keyframes spinners-angular-round-outlined{35%,65%{r:2px;stroke-width:4}}@keyframes spinners-angular-round-outlined{35%,65%{r:2px;stroke-width:4}}\n"]
            }]
    }], null, null); })();

class SpinnerRoundOutlinedModule {
}
SpinnerRoundOutlinedModule.ɵfac = function SpinnerRoundOutlinedModule_Factory(t) { return new (t || SpinnerRoundOutlinedModule)(); };
SpinnerRoundOutlinedModule.ɵmod = /*@__PURE__*/ ɵngcc0.ɵɵdefineNgModule({ type: SpinnerRoundOutlinedModule });
SpinnerRoundOutlinedModule.ɵinj = /*@__PURE__*/ ɵngcc0.ɵɵdefineInjector({ imports: [[BrowserModule]] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(SpinnerRoundOutlinedModule, [{
        type: NgModule,
        args: [{
                imports: [BrowserModule],
                declarations: [SpinnerRoundOutlinedComponent],
                exports: [SpinnerRoundOutlinedComponent]
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(SpinnerRoundOutlinedModule, { declarations: function () { return [SpinnerRoundOutlinedComponent]; }, imports: function () { return [BrowserModule]; }, exports: function () { return [SpinnerRoundOutlinedComponent]; } }); })();

/**
 * Generated bundle index. Do not edit.
 */

export { SpinnerRoundOutlinedComponent, SpinnerRoundOutlinedModule };
