// spinners-angular/__ivy_ngcc__/fesm2015/spinners-angular-spinner-dotted.js
import { Component, NgModule } from '@angular/core';
import { SpinnerComponent } from 'spinners-angular/spinners-utils';
import { BrowserModule } from '@angular/platform-browser';

import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/common';

function SpinnerDottedComponent__svg_svg_0__svg_circle_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵnamespaceSVG();
    ɵngcc0.ɵɵelement(0, "circle", 3);
} if (rf & 2) {
    const i_r3 = ctx.index;
    const ctx_r1 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵproperty("ngStyle", ctx_r1.circleStyle(i_r3));
    ɵngcc0.ɵɵattribute("r", 3 * (ctx_r1.thickness / 100));
} }
function SpinnerDottedComponent__svg_svg_0_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵnamespaceSVG();
    ɵngcc0.ɵɵelementStart(0, "svg", 1);
    ɵngcc0.ɵɵtemplate(1, SpinnerDottedComponent__svg_svg_0__svg_circle_1_Template, 1, 2, "circle", 2);
    ɵngcc0.ɵɵelement(2, "circle", 3);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵproperty("ngStyle", ctx_r0.svgStyle);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngForOf", ctx_r0.coords);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngStyle", ctx_r0.centerStyle);
    ɵngcc0.ɵɵattribute("r", 6 * (ctx_r0.thickness / 100));
} }
class SpinnerDottedComponent extends SpinnerComponent {
    constructor() {
        super(...arguments);
        this.coords = [
            { x: 22, y: -20 },
            { x: 29, y: 0 },
            { x: 22, y: 20 },
            { x: 0, y: 30 },
            { x: -23, y: 20 },
            { x: -30, y: 0 },
            { x: -23, y: -20 },
            { x: 0, y: -30 },
        ];
    }
    get duration() {
        return 200 / this.speed;
    }
    get centerStyle() {
        return !this.still
            ? {
                animation: `spinners-angular-dotted-center ${this.duration}s ease-out infinite`,
                transformOrigin: 'center',
            }
            : { display: 'none' };
    }
    circleStyle(i) {
        return Object.assign({ transform: `translate(${this.coords[i].x}px, ${this.coords[i].y}px)` }, (!this.still
            ? {
                animation: `spinners-angular-dotted-shrink ${this.duration}s cubic-bezier(0, 0.9, 0, 0.9) ${(this.duration / 20) * i}s infinite`
            }
            : {}));
    }
}
SpinnerDottedComponent.ɵfac = /*@__PURE__*/ function () { let ɵSpinnerDottedComponent_BaseFactory; return function SpinnerDottedComponent_Factory(t) { return (ɵSpinnerDottedComponent_BaseFactory || (ɵSpinnerDottedComponent_BaseFactory = ɵngcc0.ɵɵgetInheritedFactory(SpinnerDottedComponent)))(t || SpinnerDottedComponent); }; }();
SpinnerDottedComponent.ɵcmp = /*@__PURE__*/ ɵngcc0.ɵɵdefineComponent({ type: SpinnerDottedComponent, selectors: [["sa-spinner-dotted"]], features: [ɵngcc0.ɵɵInheritDefinitionFeature], decls: 1, vars: 1, consts: [["fill", "none", "viewBox", "0 0 66 66", 3, "ngStyle", 4, "ngIf"], ["fill", "none", "viewBox", "0 0 66 66", 3, "ngStyle"], ["cx", "33", "cy", "33", "fill", "currentColor", 3, "ngStyle", 4, "ngFor", "ngForOf"], ["cx", "33", "cy", "33", "fill", "currentColor", 3, "ngStyle"]], template: function SpinnerDottedComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵtemplate(0, SpinnerDottedComponent__svg_svg_0_Template, 3, 4, "svg", 0);
    } if (rf & 2) {
        ɵngcc0.ɵɵproperty("ngIf", ctx.enabled);
    } }, directives: [ɵngcc1.NgIf, ɵngcc1.NgStyle, ɵngcc1.NgForOf], styles: ["@-webkit-keyframes spinners-angular-dotted-center{0%,15%,85%,to{transform:scale(0)}40%,50%{transform:scale(1)}84%{transform:scale(.45)}}@keyframes spinners-angular-dotted-center{0%,15%,85%,to{transform:scale(0)}40%,50%{transform:scale(1)}84%{transform:scale(.45)}}@-webkit-keyframes spinners-angular-dotted-shrink{50%{transform:translate(0)}}@keyframes spinners-angular-dotted-shrink{50%{transform:translate(0)}}"] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(SpinnerDottedComponent, [{
        type: Component,
        args: [{
                selector: 'sa-spinner-dotted',
                template: "<svg *ngIf=\"enabled\" fill=\"none\" [ngStyle]=\"svgStyle\" viewBox=\"0 0 66 66\">\r\n  <circle\r\n    *ngFor=\"let c of coords; index as i\"\r\n    cx=\"33\"\r\n    cy=\"33\"\r\n    fill=\"currentColor\"\r\n    [attr.r]=\"3 * (thickness / 100)\"\r\n    [ngStyle]=\"circleStyle(i)\"\r\n  />\r\n  <circle\r\n    cx=\"33\"\r\n    cy=\"33\"\r\n    fill=\"currentColor\"\r\n    [attr.r]=\"6 * (thickness / 100)\"\r\n    [ngStyle]=\"centerStyle\"\r\n  />\r\n</svg>\r\n",
                styles: ["@-webkit-keyframes spinners-angular-dotted-center{0%,15%,85%,to{transform:scale(0)}40%,50%{transform:scale(1)}84%{transform:scale(.45)}}@keyframes spinners-angular-dotted-center{0%,15%,85%,to{transform:scale(0)}40%,50%{transform:scale(1)}84%{transform:scale(.45)}}@-webkit-keyframes spinners-angular-dotted-shrink{50%{transform:translate(0)}}@keyframes spinners-angular-dotted-shrink{50%{transform:translate(0)}}\n"]
            }]
    }], null, null); })();

class SpinnerDottedModule {
}
SpinnerDottedModule.ɵfac = function SpinnerDottedModule_Factory(t) { return new (t || SpinnerDottedModule)(); };
SpinnerDottedModule.ɵmod = /*@__PURE__*/ ɵngcc0.ɵɵdefineNgModule({ type: SpinnerDottedModule });
SpinnerDottedModule.ɵinj = /*@__PURE__*/ ɵngcc0.ɵɵdefineInjector({ imports: [[BrowserModule]] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(SpinnerDottedModule, [{
        type: NgModule,
        args: [{
                imports: [BrowserModule],
                declarations: [SpinnerDottedComponent],
                exports: [SpinnerDottedComponent]
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(SpinnerDottedModule, { declarations: function () { return [SpinnerDottedComponent]; }, imports: function () { return [BrowserModule]; }, exports: function () { return [SpinnerDottedComponent]; } }); })();

/**
 * Generated bundle index. Do not edit.
 */

export { SpinnerDottedComponent, SpinnerDottedModule };
