// spinners-angular/__ivy_ngcc__/fesm2015/spinners-angular-spinner-circular-fixed.js
import { Component, NgModule } from '@angular/core';
import { SpinnerComponentWithSecondaryColor } from 'spinners-angular/spinners-utils';
import { BrowserModule } from '@angular/platform-browser';

import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/common';

function SpinnerCircularFixedComponent__svg_svg_0_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵnamespaceSVG();
    ɵngcc0.ɵɵelementStart(0, "svg", 1);
    ɵngcc0.ɵɵelement(1, "circle", 2);
    ɵngcc0.ɵɵelement(2, "circle", 3);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵproperty("ngStyle", ctx_r0.svgStyle);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵattribute("stroke", ctx_r0.secondaryColor)("stroke-width", ctx_r0.strokeWidth);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngStyle", ctx_r0.dashStyle);
    ɵngcc0.ɵɵattribute("stroke-width", ctx_r0.strokeWidth);
} }
class SpinnerCircularFixedComponent extends SpinnerComponentWithSecondaryColor {
    get strokeWidth() {
        return 4 * (this.thickness / 100);
    }
    get dashStyle() {
        return Object.assign({ color: this.color }, (!this.still
            ? { animation: `spinners-angular-circular-fixed ${140 / this.speed}s linear infinite` }
            : {}));
    }
}
SpinnerCircularFixedComponent.ɵfac = /*@__PURE__*/ function () { let ɵSpinnerCircularFixedComponent_BaseFactory; return function SpinnerCircularFixedComponent_Factory(t) { return (ɵSpinnerCircularFixedComponent_BaseFactory || (ɵSpinnerCircularFixedComponent_BaseFactory = ɵngcc0.ɵɵgetInheritedFactory(SpinnerCircularFixedComponent)))(t || SpinnerCircularFixedComponent); }; }();
SpinnerCircularFixedComponent.ɵcmp = /*@__PURE__*/ ɵngcc0.ɵɵdefineComponent({ type: SpinnerCircularFixedComponent, selectors: [["sa-spinner-circular-fixed"]], features: [ɵngcc0.ɵɵInheritDefinitionFeature], decls: 1, vars: 1, consts: [["fill", "none", "viewBox", "0 0 66 66", 3, "ngStyle", 4, "ngIf"], ["fill", "none", "viewBox", "0 0 66 66", 3, "ngStyle"], ["cx", "33", "cy", "33", "fill", "none", "r", "28"], ["cx", "33", "cy", "33", "fill", "none", "r", "28", "stroke", "currentColor", "stroke-dasharray", "40, 134", "stroke-dashoffset", "325", "stroke-linecap", "round", 3, "ngStyle"]], template: function SpinnerCircularFixedComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵtemplate(0, SpinnerCircularFixedComponent__svg_svg_0_Template, 3, 5, "svg", 0);
    } if (rf & 2) {
        ɵngcc0.ɵɵproperty("ngIf", ctx.enabled);
    } }, directives: [ɵngcc1.NgIf, ɵngcc1.NgStyle], styles: ["@-webkit-keyframes spinners-angular-circular-fixed{0%{stroke-dashoffset:325}to{stroke-dashoffset:151}}@keyframes spinners-angular-circular-fixed{0%{stroke-dashoffset:325}to{stroke-dashoffset:151}}"] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(SpinnerCircularFixedComponent, [{
        type: Component,
        args: [{
                selector: 'sa-spinner-circular-fixed',
                template: "<svg *ngIf=\"enabled\" fill=\"none\" [ngStyle]=\"svgStyle\" viewBox=\"0 0 66 66\">\r\n  <circle\r\n    cx=\"33\"\r\n    cy=\"33\"\r\n    fill=\"none\"\r\n    r=\"28\"\r\n    [attr.stroke]=\"secondaryColor\"\r\n    [attr.stroke-width]=\"strokeWidth\"\r\n  />\r\n  <circle\r\n    cx=\"33\"\r\n    cy=\"33\"\r\n    fill=\"none\"\r\n    r=\"28\"\r\n    stroke=\"currentColor\"\r\n    stroke-dasharray=\"40, 134\"\r\n    stroke-dashoffset=\"325\"\r\n    stroke-linecap=\"round\"\r\n    [attr.stroke-width]=\"strokeWidth\"\r\n    [ngStyle]=\"dashStyle\"\r\n  />\r\n</svg>\r\n",
                styles: ["@-webkit-keyframes spinners-angular-circular-fixed{0%{stroke-dashoffset:325}to{stroke-dashoffset:151}}@keyframes spinners-angular-circular-fixed{0%{stroke-dashoffset:325}to{stroke-dashoffset:151}}\n"]
            }]
    }], null, null); })();

class SpinnerCircularFixedModule {
}
SpinnerCircularFixedModule.ɵfac = function SpinnerCircularFixedModule_Factory(t) { return new (t || SpinnerCircularFixedModule)(); };
SpinnerCircularFixedModule.ɵmod = /*@__PURE__*/ ɵngcc0.ɵɵdefineNgModule({ type: SpinnerCircularFixedModule });
SpinnerCircularFixedModule.ɵinj = /*@__PURE__*/ ɵngcc0.ɵɵdefineInjector({ imports: [[BrowserModule]] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(SpinnerCircularFixedModule, [{
        type: NgModule,
        args: [{
                imports: [BrowserModule],
                declarations: [SpinnerCircularFixedComponent],
                exports: [SpinnerCircularFixedComponent]
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(SpinnerCircularFixedModule, { declarations: function () { return [SpinnerCircularFixedComponent]; }, imports: function () { return [BrowserModule]; }, exports: function () { return [SpinnerCircularFixedComponent]; } }); })();

/**
 * Generated bundle index. Do not edit.
 */

export { SpinnerCircularFixedComponent, SpinnerCircularFixedModule };
